app/code/Magento/Checkout/Model/GuestPaymentInformationManagement.php

```diff
@@ -108,12 +114,15 @@ public function savePaymentInformationAndPlaceOrder(
             try {
                 $orderId = $this->cartManagement->placeOrder($cartId);
             } catch (\Magento\Framework\Exception\LocalizedException $e) {
+                $quoteIdMask = $this->MaskedQuoteIdToQuoteIdInterface->create()->load($cartId, 'masked_id');
```

Comment:
Did you check this code? Are you sure it will work?
Please check how to use MaskedQuoteIdToQuoteIdInterface. It doesn't have a method "create".
MaskedQuoteIdToQuoteId which implements MaskedQuoteIdToQuoteIdInterface has methos "execute" and check returning type. It doesn't return  QuoteIdMask model.